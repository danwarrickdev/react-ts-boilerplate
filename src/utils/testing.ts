// @ts-nocheck
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/react';
import preview from '../../.storybook/preview';

export const testStories = (Stories: any) => {
  const wrapper = preview?.decorators?.[0];
  const testCases = Object.entries(composeStories(Stories));
  /* eslint-disable no-undef */
  test.each(testCases)('Renders %s story', async (_StoryName, Story) => {
    const tree = await render(wrapper(Story, Story));
    /* eslint-disable no-undef */
    expect(tree.baseElement).toMatchSnapshot();
  });
};
