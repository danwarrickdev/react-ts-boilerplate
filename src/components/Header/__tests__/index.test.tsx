import { testStories } from '../../../utils/testing';
import * as Stories from '../index.stories';

describe('Header', () => {
  describe('All stories should render successfully', () => {
    testStories(Stories);
  });
});
