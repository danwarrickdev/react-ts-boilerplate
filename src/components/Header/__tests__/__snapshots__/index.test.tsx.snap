// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header All stories should render successfully Renders Primary story 1`] = `
.emotion-0 {
  min-height: 100vh;
}

<body>
  <div>
    <div
      class="emotion-0"
    >
      <div
        style="height: 100vh; width: 100vw;"
      >
        <div>
          <nav>
            <a
              aria-current="page"
              class="active"
              data-discover="true"
              href="/"
            >
              Home
            </a>
            <a
              class=""
              data-discover="true"
              href="/sample"
            >
              Sample
            </a>
          </nav>
          <button>
            Light Mode
          </button>
          <button>
            Dark Mode
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
